### ISSUE TITLE
Support relationships between pages/posts/publications

### ISSUE THREAD

**Issue by Toxaris** [ps-mr/website#30]
Each post/page/publication should support a list of related posts/pages/publications. For example:

 - a page about a research project should have a list of related publications, related thesis projects, and related team members
 - a page about a team member should have a list of related publications, related thesis projects, and related research projects

Hmm, being related appears to be a symmetric relation. Therefore, I propose that we support `related: id-of-related-item` headers in all posts and pages, and `related=...` attributes in bibtex files. We then compute the symmetric closure of these entries and make them available to the Liquid templates.

**Comment by Blaisorblade** [comment]
I thought this make sense, but I realized I have a hard time thinking of a good template using this — both for Jekyll templates, and for individual pages.

Tautologically, "related" seems too coarse a relation: Would you lump together all related things in a list and have any template dump it on the page? If we want to have support for this, we probably want *multiple* relations. However, even that seems an excessively "waterfall" design. Maybe we want concrete examples of what to abstract over.

In fact, I think we want good old links. However, we might need to generate them. I guess I'd like more tags like `{% member id %}`, one for each category you name — so we'd additionally need `{% resProj id %}`, `{% thesisProj id %}`, and `{% paper id %}`. Current theses topics use `{% member id %}` and ad-hoc forms of `{% thesisProj id %}`; I've decided against linking to related papers, since I hope the project home can be a more gentle intro while the paper is too scary. Also, no lists of related things.

However, automation sounds more plausible for a few cases.
- It would be cool if publications automatically linked to authors who have a profile here.
- Authors might want a publication list on their profile page. A topic-sorted version (written by hand) also makes sense, but a general list sounds a good default.

**Comment by Toxaris** [comment]
No, a page (layout) would not (have to) show all related items in one big list, but could filter and/or sort them by the attributes of the related item. For example, a research project could show a lists of open thesis projects, a list of finished thesis projects, and a list of publications.

I guess that in practice, the semantics of a relation edge is uniquely defined by the attributes of the source and target pages. Counter examples welcome.

PS. The goal of this design is to implement something generic but not too complicated in Ruby (page lookup and symmetric closure) and then build on top in Liquid (presentation of the lists of related pages). This way, most website editors can tweak the data model a bit without touching the rather fragile Ruby code.

**Comment by Blaisorblade** [comment]
> No, a page (layout) would not (have to) show all related items in one big list, but could filter and/or sort them by the attributes of the related item. For example, a research project could show a lists of open thesis projects, a list of finished thesis projects, and a list of publications.

Ah, then I retract my opposition.

> I guess that in practice, the semantics of a relation edge is uniquely defined by the attributes of the source and target pages. Counter examples welcome.

Hm. Either fragile or insightful — right now, I'm guessing insightful. :+1:

**Comment by Toxaris** [comment]
I'm tempted to support free-form tags on relation edges, as well as a specification language that can express closure properties (such as "symmetric closure", but also others). Together with a user-defined syntactic sugar, that would make a nice language to model the website's content in. Oh wait, that's Ensō. ;)

**Comment by Toxaris** [comment]
Missing features:

 - relations involving publications
 - relations involving entities not on our website
